library/src/test/java/com/paypal/messages/logger/TrackingPayloadTest.kt

```diff
@@ -32,21 +32,38 @@ class TrackingPayloadTest {
 		libraryVersion = libraryVersion,
 		components = components,
 	)
-	
+
+	private val cloudWrappedEvent = CloudEvent(
+		specversion = "1.0",
+		type = "com.paypal.credit.upstream-presentment.v1",
+		source = "urn:paypal:event-src:v1:android:messages",
+		datacontenttype = "application/json",
+		dataschema = "ppaas:events.credit.FinancingPresentmentAsyncAPISpecification/v1/schema/json/credit_upstream_presentment_event.json",
+		data = trackingPayload,
+	)
+
 	@Test
 	fun testConstructor() {
-		assertEquals(clientId, trackingPayload.clientId)
-		assertEquals(merchantId, trackingPayload.merchantId)
-		assertEquals(partnerAttributionId, trackingPayload.partnerAttributionId)
-		assertEquals(merchantProfileHash, trackingPayload.merchantProfileHash)
-		assertEquals(deviceId, trackingPayload.deviceId)
-		assertEquals(sessionId, trackingPayload.sessionId)
-		assertEquals(instanceId, trackingPayload.instanceId)
-		assertEquals(integrationName, trackingPayload.integrationName)
-		assertEquals(integrationType, trackingPayload.integrationType)
-		assertEquals(integrationVersion, trackingPayload.integrationVersion)
-		assertEquals(libraryVersion, trackingPayload.libraryVersion)
-		assertEquals(components, trackingPayload.components)
+		assertEquals(cloudWrappedEvent.specversion, "1.0")
```

Comment:
I'd like every file to have its own test file, so let's create a CloudEventTest and move some of this over. The new file doesn't have to deep test trackingPayload since that'll be tested here. We can just test for cloudEvent source, type, specversion (now that I'm seeing this, let's rename it to specVersion), datacontenttype(rename to dataContentType) and dataschema (rename to dataSchema).

The new file should also test for the CloudEvent's ability to serialize. You can definitely copy a lot of what's here. It's redundant but I'd rather be redundant at this point.

Reply:
Unfortunately specversion, datacontenttype, and dataschema are cased this way in the PPaaS cloud event usage spec so I had to keep it for the data class so its sent correctly.

Reply:
You can use the @SerializedName annotation to name things with camel case but still send them in the way the API requires. 

```
@SerializedName("specversion")
val specVersion: String = "1.0",
```